metaflow/plugins/aip/aip_cli.py

```diff
@@ -417,14 +417,14 @@ def _echo_workflow_run(
     flow_name: str, kubernetes_namespace: str, obj, workflow_manifest: Dict[str, Any]
 ):
     argo_workflow_name = workflow_manifest["metadata"]["name"]
-    metaflow_run_id = to_metaflow_run_id(argo_workflow_name)
-    metaflow_ui_url = run_id_to_metaflow_url(flow_name, argo_workflow_name)
+    argo_workflow_uid = workflow_manifest["metadata"]["uid"]
```

Comment:
looks good, that is exactly how argo archives historical flows and distinguishes them.